chromium_src/net/base/network_isolation_key.cc

```diff
@@ -0,0 +1,19 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "net/base/network_isolation_key.h"
+
+#include "../../../../net/base/network_isolation_key.cc"
+
+namespace net {
+
+const base::Optional<url::Origin>&
+NetworkIsolationKey::GetEffectiveTopFrameOrigin() const {
+  DCHECK_EQ(original_top_frame_origin_.has_value(),
+            top_frame_origin_.has_value());
+  return top_frame_origin_;
```

Comment:
this seems extremely risky to me and I don't think we should make this change. Why wouldn't we just use the method as-is?

Reply:
Replied to this in the PR here: https://github.com/brave/brave-core/pull/7199#issuecomment-731553278

TL;DR: We need it because the currently available getter `GetTopFrameOrigin()` returns the original value passed to the constructor, unprocessed. And for the CircuitIsolationKey we need the value after having been processed with GetDomainAndRegistry(), which is different from the original value passed to the constructor.

Reply:
I was skimming the NIK sources and found `NetworkIsolationKey::CreateWithNewFrameOrigin`. Wondering, would it be possible to avoid override with a hack like this:
```
net::NetworkIsolationKey network_isolation_key(url_origin, url_origin);
net::NetworkIsolationKey real_nik = network_isolation_key.CreateWithNewFrameOrigin(url_origin);
```
looks like it does the trick?
@mariospr

Reply:
@iefremov Good point! I did look into the very same function on Friday, but that doesn't work unfortunately in the code we have in master because the implementation of that method [still uses `top_frame_origin_` the create the second key](https://chromium.googlesource.com/chromium/src.git/+/refs/tags/87.0.4280.67/net/base/network_isolation_key.cc#93):
```
  NetworkIsolationKey NetworkIsolationKey::CreateWithNewFrameOrigin(
      const url::Origin& new_frame_origin) const {
    if (!top_frame_origin_)
      return NetworkIsolationKey();
    NetworkIsolationKey key(top_frame_origin_.value(), new_frame_origin);
    key.opaque_and_non_transient_ = opaque_and_non_transient_;
    return key;
  }
```
This said, I've just noticed that it seems that such `chromium_src` override won't be needed anymore once we rebase this PR on top of Chromium 88, because [`GetTopFrameSite()` does no longer return the original value in there](https://chromium.googlesource.com/chromium/src.git/+/refs/tags/88.0.4324.11/net/base/network_isolation_key.h#113), 
but [the one that has been processed right on the constructor](https://chromium.googlesource.com/chromium/src.git/+/refs/tags/88.0.4324.11/net/base/network_isolation_key.cc#235).

Thus, consider this chromium_src as a temporary thing, will go away from Cr88 on!

Reply:
but isn't `top_frame_origin_` is what we actually want?.. (you return it in the newly added getter)
anyway, I'm fine with leaving it as is till cr88 - probably worth filing an issue?

Reply:
Hmm.. I think you're right, sorry. I got confused with `original` and `origin` and thought it was returning `original_top_frame_origin_`. Now I'm not sure why this trick did not work when I tried it on Friday, maybe I did something else wrong and blamed it on this by mistake. My bad, sorry for the confusion.

As for filing an issue, just one comment on that this patch is only a thing in Cr87 and will go away once we rebase this fix on top of Cr88, where `GetTopFrameSite()` will return what we want already and we will be able to drop it. Still, I think filing an issue makes sense because we'll need to remember to drop this either way, or we'll risk leaving this override or the hack around for longer, so here it is: https://github.com/brave/brave-browser/issues/12869

Thanks, and apologies again for the confusion!